lib/public/systemtag/isystemtagobjectmapper.php

```
@@ -0,0 +1,106 @@
+<?php
+/**
+ * @author Vincent Petry <pvince81@owncloud.com>
+ *
+ * @copyright Copyright (c) 2015, ownCloud, Inc.
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCP\SystemTag;
+
+/**
+ * Public interface to access and manage system-wide tags.
+ *
+ * @since 9.0.0
+ */
+interface ISystemTagObjectMapper {
+
+	/**
+	 * Get a list of tag ids for the given object ids.
+	 *
+	 * This returns an array that maps object id to tag ids
+	 * [
+	 *   1 => array('id1', 'id2'),
+	 *   2 => array('id3', 'id2'),
+	 *   3 => array('id5'),
+	 *   4 => array()
+	 * ]
+	 *
+	 * Untagged objects will have an empty array associated.
+	 *
+	 * @param string|array $objIds object ids
+	 * @param string $objectType object type
+	 *
+	 * @return array with object id as key and an array
+	 * of tag ids as value
+	 *
+	 * @since 9.0.0
+	 */
+	public function getTagIdsForObjects($objIds, $objectType);
+
+	/**
+	 * Get a list of objects tagged with $tagIds.
+	 *
+	 * Throws an exception if the tag could not be found.
+	 *
+	 * @param string|array $tagIds Tag id or array of tag ids.
+	 * @param string $objectType object type
+	 *
+	 * @return string[] array of object ids or empty array if none found
+	 *
+	 * @since 9.0.0
+	 */
+	public function getObjectIdsForTags($tagIds, $objectType);
+
+	/**
+	 * Assign the given tags to the given object.
```

assign and unassign should `@throw` as well when the tag ID is invalid